sdv/constraints/tabular.py

```diff
@@ -892,19 +892,19 @@ def is_valid(self, table_data):
         middle = table_data[self.middle_column_name]
         high = table_data[self.high_column_name]
 
-        satisfy_low_bound = np.logical_or(
-            self._operator(low, middle),
-            np.isnan(low),
-        )
-        satisfy_high_bound = np.logical_or(
-            self._operator(middle, high),
-            np.isnan(high),
-        )
+        list_columns_nans = [self.low_column_name, self.middle_column_name, self.high_column_name]
+        nan_column = compute_nans_column(table_data, list_columns_nans)
+        validate_column = pd.Series([True]*len(table_data), index=table_data.index)
+        if nan_column is None:
+            validate_column = self._operator(low, middle) * self._operator(middle, high)
+        else:
+            validate_column.loc[nan_column == 'None'] = self._operator(
+                low, middle) * self._operator(middle, high)
+            validate_column.loc[nan_column == self.low_column_name] = self._operator(middle, high)
+            validate_column.loc[nan_column == self.middle_column_name] = self._operator(low, high)
+            validate_column.loc[nan_column == self.high_column_name] = self._operator(low, middle)
```

Comment:
I think it might be simpler to do something like 
```python
low_lt_middle = self._operator(low, middle) | np.isnan(low) | np.isnan(middle)
middle_lt_high = self._operator(middle, high) | np.isnan(middle) | np.isnan(high)
low_lt_high = self._operator(low, high) | np.isnan(low) | np.isnan(high)
return low_lt_middle & middle_lt_high & low_lt_high
```
You might have to test which is more performant though